Jenkinsfile

```diff
@@ -19,12 +19,30 @@ def mavenEnv(Map params = [:], Closure body) {
       timeout(120) {
         withChecks(name: 'Tests', includeStage: true) {
           infra.withArtifactCachingProxy {
-            withEnv([
-              'JAVA_HOME=/opt/jdk-' + params['jdk'],
-              "MAVEN_ARGS=${env.MAVEN_ARGS != null ? MAVEN_ARGS : ''} -B -ntp -Dmaven.repo.local=${WORKSPACE_TMP}/m2repo"
-            ]) {
-              body()
-            }
+            def mavenRepoPath = "/tmp/.m2-cache-${params['cacheKey']}"
```

Comment:
What is the rationale behing using `/tmp` ? I see the comment below but I'm not sure to understand it at all? It feels like the cache is shared between branches and builds: is that correct?
Couldn't we use the `${WORKSPACE_TMP}` variable?

> // using a fixed path for Maven cache instead of the normal workspace pattern
> // due to the cache name being calculated from the path
> // this prevents seeding working if you use the normal pattern

Reply:
We can't use WORKSPACE_TMP as that is dynamic and would prevent seeding the cache on PRs (it includes the folder name, e.g. PR-22), the cache path needs to be static as thats used for calculating the cache name.

We could also use a folder inside of the workspace, or calculate the relative path to `WORKSPACE`.

I didn't think it matters what since these are ephemeral containers.

Reply:
Oh I see, it is on the agent side. Then LGTM, except that using `/tmp` could lead to issue on different nodes as containerd might mount `/tmp` into ramfs, local dir, or overlay depending on the cluster node setups.

WDYT about using the environment variable `CACHE_DIR` or fall backing to `/tmp` if undefined or empty?
=> With this, we (admin) could then update the pod templates to provide a local emptydir volume mounted in `$CACHE_DIR` to avoid tmpfs (small size) or overlay (small performances)

Reply:
```suggestion
            def cacheBaseDir = env.JOBCACHER_CACHE_DIR ?: "/tmp"
            def mavenRepoPath = "${cacheBaseDir}/.m2-cache-${params['cacheKey']}"
```